### PR TITLE
Add display current ping to world hopper

### PR DESCRIPTION
**Pull request by gregg1494**
ping

Moved my other PR into world hopper as it was redundant to have its own plugin. The current ping time is every second. It can be changed if that is too frequent. 

If display current ping is checked it will also update the world hopper ping panel as well, so they are consistent.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+public class WorldHopperOverlay extends Overlay
+{
+	private static final int Y_OFFSET = 11;
+	private static final int VALUE_X_OFFSET = 1;
```
Comment (xDemoN) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: Is there a reason "VALUE_" is here?

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java [hunk 2]
```diff
@@ -100,13 +102,17 @@
 	private static final int REFRESH_THROTTLE = 60_000;  // ms
 	private static final int TICK_THROTTLE = (int) Duration.ofMinutes(10).toMillis();
 
+	public static final int CURR_WORLD_PING_TIMER = 1;
```
Comment (xDemoN) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java]: "Curr" is a really odd abbreviation

Comment (gregg1494) [reply runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java]: fixed

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java (hunk 1 above)
Comment (gregg1494) [reply runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: fixed

**Comment by Adam-** [comment]
I would make `pingWorlds` skip the current world then if displayPings is on.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+public class WorldHopperOverlay extends Overlay
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperOverlay.java]: remove public

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java [hunk 4]
```diff
@@ -761,6 +811,10 @@ private void pingWorlds()
 
 		for (World world : worldResult.getWorlds())
 		{
+			if (config.displayPing() && client.getWorld() == world.getId())
+			{
+				continue;
+			}
 			int ping = Ping.ping(world);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java]: add empty line above

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java [hunk 5]
```diff
@@ -769,4 +823,17 @@ private void pingWorlds()
...
 	}
+
+	private void pingWorld()
+	{
+		if (worldResult == null || !config.displayPing())
+		{
+			return;
+		}
+
+		World currentWorld = worldResult.findWorld(client.getWorld());
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java]: final

**Comment by Adam-** [comment]
@deathbeam I moved this to ping the current world on the main pinger executor, and then split up the existing world ping code to ping worlds incrementally so that the scheduler can schedule in the current world pings too. Otherwise, it was not updating the current ping for 1-2 mins while the pings were being updated.

**Comment by Adam-** [comment]
This is probably okay now from a code perspective. However it looks like it draws the ping time over the logout button - is this intentional?
image

**Comment by gregg1494** [comment]
> This is probably okay now from a code perspective. However it looks like it draws the ping time over the logout button - is this intentional?
> image

image

Ya it is intentional. The fps is also displayed over the logout button, if it becomes a problem we could look at moving them.

**Comment by gregg1494** [comment]
image

**Comment by Adam-** [comment]
Do we want to make it draw it to the right if there is no X logout button? I think it depends on sidestones arrangement.

**Comment by deathbeam** [comment]
Yes. It should just check if the logout button exists, and if yes, offset by its width (like ClientUI is doing atm with sidebar button).

**Pull request merged**